### PR TITLE
BoxesActionsAndTable: Connect setSelectedBoxes to Checkboxes in Boxes Tables

### PR DESCRIPTION
**Pull request by vahidbazzaz**
- wire up setSelectedBoxes to rows that checked
- implement test case 4.8.2.1

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[2.0 BoxesActionsAndTable: connect setSelectedBoxes to checkboxes in Boxes Tables](https://trello.com/c/9m0l9ZC1/1302-20-boxesactionsandtable-connect-setselectedboxes-to-checkboxes-in-boxes-tables)

File: front/src/views/Boxes/components/BoxesTable.tsx [hunk 1]
```diff
@@ -34,6 +35,7 @@ interface IBoxesTableProps {
   actionButtons: React.ReactNode[];
   columnSelector: React.ReactNode;
   onBoxRowClick: (labelIdentified: string) => void;
+  setSelectedBoxes: (rows: Row<any>[]) => void;
```
Comment (HaGuesto) [inline front/src/views/Boxes/components/BoxesTable.tsx:38]: Is there a reason why you did not define it like this?
`setSelectedBoxes: (rows: BoxRow[]) => void`

File: front/src/views/Boxes/components/BoxesActionsAndTable.tsx [hunk 2]
```diff
@@ -51,14 +50,16 @@ function BoxesActionsAndTable({
       },
     },
   ]);
+
   const onMoveBoxes = useCallback(
     (locationId: string) =>
       moveBoxesAction.moveBoxes(
-        selectedBoxes.map((box) => box.values.labelIdentifier),
+        selectedBoxes.map((box) => (box as unknown as BoxRow).labelIdentifier),
```
Comment (HaGuesto) [inline front/src/views/Boxes/components/BoxesActionsAndTable.tsx:57]: that type casting looks weird.

File: front/src/views/Boxes/components/BoxesActionsAndTable.tsx [hunk 3]
```diff
@@ -38,7 +38,6 @@ function BoxesActionsAndTable({
     navigate(`/bases/${baseId}/boxes/${labelIdentifier}`);
 
   // --- Actions on selected Boxes
-  // TODO: pass set SelectedBoxes to checkboxes in Boxes Tables
   // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
```
Comment (HaGuesto) [inline front/src/views/Boxes/components/BoxesActionsAndTable.tsx:41]: these eslint ignores should be removed too

**Review by HaGuesto — APPROVED** [review]
looks good @vahid. Happy to merge it. We can fix little things here and there at the end.

**Pull request merged**